openpype/modules/shotgrid/plugins/publish/collect_shotgrid_session.py

```diff
@@ -0,0 +1,131 @@
+                )
+            )
+            return
+
+        set_shotgrid_certificate(certificate_path)
+        self.log.info("Set Certificate: {}".format(certificate_path))
+
+        avalon_project = os.getenv("AVALON_PROJECT")
+
+        shotgrid_settings = get_shotgrid_settings(avalon_project)
+        self.log.info("shotgrid settings: {}".format(shotgrid_settings))
+        shotgrid_servers_settings = get_shotgrid_servers()
+        self.log.info(
+            "shotgrid_servers_settings: {}".format(shotgrid_servers_settings)
+        )
+
+        shotgrid_server = shotgrid_settings.get("shotgrid_server", "")
+        if not shotgrid_server:
+            self.log.error(
+                "No Shotgrid server found, please choose a credential"
+                "in script name and script key in OpenPype settings"
+            )
+
+        shotgrid_server_setting = shotgrid_servers_settings.get(
+            shotgrid_server, {}
+        )
+        shotgrid_url = shotgrid_server_setting.get("shotgrid_url", "")
+
+        shotgrid_script_name = shotgrid_server_setting.get(
+            "shotgrid_script_name", ""
+        )
+        shotgrid_script_key = shotgrid_server_setting.get(
+            "shotgrid_script_key", ""
+        )
+        if not shotgrid_script_name and not shotgrid_script_key:
+            self.log.error(
+                "No Shotgrid api credential found, please enter "
+                "script name and script key in OpenPype settings"
+            )
+
+        login = get_login() or os.getenv("OPENPYPE_SG_USER")
+
+        if not login:
+            self.log.error(
+                "No Shotgrid login found, please "
+                "login to shotgrid withing openpype Tray"
+            )
+
+        session = shotgun_api3.Shotgun(
+            base_url=shotgrid_url,
+            script_name=shotgrid_script_name,
+            api_key=shotgrid_script_key,
+            sudo_as_login=login,
+        )
+
+        try:
+            session.preferences_read()
+        except AuthenticationFault:
+            raise ValueError(
+                "Could not connect to shotgrid {} with user {}".format(
+                    shotgrid_url, login
+                )
+            )
+
+        self.log.info(
+            "Logged to shotgrid {} with user {}".format(shotgrid_url, login)
+        )
+        context.data["shotgridSession"] = session
+        context.data["shotgridUser"] = login
+
+
+def get_shotgrid_certificate():
+    shotgun_api_path = os.path.dirname(shotgun_api3.__file__)
+    return os.path.join(shotgun_api_path, "lib", "certifi", "cacert.pem")
+
+
+def set_shotgrid_certificate(certificate):
+    os.environ["SHOTGUN_API_CACERTS"] = certificate
+
+
+def get_shotgrid_settings(project):
+    return get_project_settings(project).get("shotgrid", {})
+
+
+def get_shotgrid_servers():
+    return (
+        get_system_settings()
+        .get("modules", {})
+        .get("shotgrid", {})
+        .get("shotgrid_settings", {})
+    )
```

Comment:
Should this be removed in favor of the [`settings.get_shotgrid_servers()`](https://github.com/pypeclub/OpenPype/pull/2921/files#diff-e6cac9cfcb7eb34534c06e605a8e47ed8dcb69ff45ac53d4af50cae0ada48889R24-R25) that is implemented?

Similarly some other functions here also appear to be duplicates of implementations elsewhere? Like `get_shotgrid_settings`?

Reply:
The duplicate was removed